rotkehlchen/api/rest.py

```diff
@@ -5431,3 +5432,49 @@ def get_historical_netvalue(
             result['last_event_identifier'] = last_event_id
 
         return api_response(_wrap_in_ok_result(result=result))
+
+    @async_api_call()
+    def get_historical_prices_per_asset(
+            self,
+            asset: Asset,
+            interval: int,
+            to_timestamp: Timestamp,
+            from_timestamp: Timestamp,
+    ) -> dict[str, Any]:
+        prices, no_prices_ts = {}, []
+        with (db := self.rotkehlchen.data.db).conn.read_ctx() as cursor:
+            main_currency = db.get_setting(cursor=cursor, name='main_currency')
+
+        processed_count = 0
+        total_intervals = (to_timestamp - from_timestamp) // interval + 1
+        # Send status update every 5% progress
+        update_threshold = max(total_intervals // 20, 1)
```

Comment:
no no no.

The updates should be done every `N` remote calls where N is configurable. Let's say every 5 or 10 calls?